test/controllers/api/v2/activation_keys_controller_test.rb

```diff
@@ -246,5 +246,12 @@ def test_remove_subscriptions_protected
         post(:remove_subscriptions, :organization_id => @organization.id, :id => @activation_key.id, :subscription_id => 123)
       end
     end
+
+    def test_remove_host_collections
+      @activation_key.update_attributes(host_collections: HostCollection.all)
+      assert(@activation_key.host_collection_ids.any?)
+      put(:remove_host_collections, organization_id: @organization.id, id: @activation_key.id, host_collection_ids: [])
+      assert_empty(@activation_key.host_collection_ids)
```

Comment:
My general philosophy is that that is a great goal, but if it makes the test extremely obtuse and/or doesn't actually test anything its a net negative for the project. 

You could possibly stub something like  activation_key#host_collection= and verify its passed with the correct values, but again I prefer for it to actually run the full action and verify the result.